src/Databases/DatabaseOnDisk.cpp

```diff
@@ -696,6 +697,18 @@ ASTPtr DatabaseOnDisk::parseQueryFromMetadata(
     auto ast = tryParseQuery(parser, pos, pos + query.size(), error_message, /* hilite = */ false,
                              "in file " + metadata_file_path, /* allow_multi_statements = */ false, 0, settings.max_parser_depth);
 
+    /// proton: starts
```

Comment:
We can enhance `tryParseQuery` to handle ClickHouse external table case since it is very special (without columns definition); the pros is there may be quite few places calling `tryParseQuery` and we will handle CH external parsing in a centralize place ?

Reply:
A very simple trick is wrapping tryParseQuery with 2 functions

```
ASTPtr tryParseQuery(...)
{
    auto ast = doTryParseQuery(...); /// existing logic
    if (ast) return ast;
    else return external table parsing..
}
```

Reply:
Actually, we don't need to change `tryParseQuery`, which should not be bound to any parser. What need to need change is to update `ParserCreateQuery` and make `PraserCreateExternalTableQuery` part of it.